packages/next/src/server/future/route-matcher-providers/dev/dev-app-page-route-matcher-provider.ts

```diff
@@ -59,6 +59,11 @@ export class DevAppPageRouteMatcherProvider extends FileCacheRouteMatcherProvide
 
     normalizeCatchAllRoutes(appPaths)
 
+    // Make sure to sort parallel routes to make the result deterministic.
+    appPaths = Object.fromEntries(
+      Object.entries(appPaths).map(([k, v]) => [k, v.sort()])
+    )
```

Comment:
noticed this was in `build` but not `dev`, so copied from [here](https://github.com/vercel/next.js/blob/canary/packages/next/src/build/entries.ts#L552-L555)

Reply:
leaning into that abstraction, is there some place where we could have added it only once?